src/wayland-eglsurface.c

```diff
@@ -1457,7 +1848,12 @@ static EGLint create_surface_stream_local(WlEglSurface *surface)
 
     wl_list_init(&surface->ctx.acquiredImages);
 
-    if (!surface->wlBufferEventQueue) {
+    /*
+     * Don't enable the buffer release thread when explicit sync is in use.
+     * In explicit sync we don't care about the delivery of release events, we
+     * only pay attention to the release points.
```

Comment:
When we're using explicit sync, __technically__ we shouldn't need either the per-image or per-surface mutexes, since we won't be using any background threads, right?

Having said that, making all of the `pthread_mutex_lock/unlock` calls conditional would be pretty messy (especially that dance at the end of `wlEglSwapBuffersWithDamageHook`) so maybe it's not worth whatever minuscule performance benefit we could gain by doing so.

Reply:
I agree I don't think it's a worthy tradeoff. The conditional mutex bits sound like a recipe for hard to find synchronization issues and keeping the two paths as identical as possible (at least for now) seems appropriate.